wrapanapi/systems/redfish.py

```diff
@@ -377,7 +389,12 @@ def get_rack(self, resource_id):
            InvalidValueException if the resource_id represents a Chassis that is
              not a rack
         """
-        return self.get_chassis(resource_id, "Rack")
+        rack = RedfishRack(self, raw=self.find(resource_id))
+
+        if rack.raw.ChassisType != "Rack":
```

Comment:
Maybe it would be a bit more intuitive to check if the raw value represents the rack before we construct one? I had something like this in mind:

    rack_data = self.find(resource_id)
    if rack_data.ChassisType != "Rack":
        ...
    return RedfishRack(self, rack_data)

Reply:
👍

Reply:
@matejart are you planning on making this change? conversation was resolved, but it looks like the older pattern is still here.

I'm not opposed to merging as-is, just let me know if you would like to push another commit.

Reply:
Thank you for this nudge, I did make the change, but had not pushed it. I've done that now.